base/deprecated.jl

```diff
@@ -496,3 +496,19 @@ function to_index{T<:Real}(A::AbstractArray{T})
     depwarn("indexing with non Integer AbstractArrays is deprecated", :to_index)
     Int[to_index_nodep(x) for x in A]
 end
+
+function float_isvalid{T<:Union(Float32,Float64)}(s::AbstractString, out::Array{T,1})
+    tf = tryparse(T, s)
+    isnull(tf) || (out[1] = get(tf))
+    !isnull(tf)
+end
```

Comment:
Unrelated question, in this function isnull is called twice, would it in
general offer a performance gain to save the value from the first call in a
local variable or is the execution time equivalent to that of fetching the
variable? Is there a definable turning point in function complexity at
which saving its output becomes beneficial? Can saving array values as
local variables offer improvement or does Julia automatically detect and
optimize reusable local values?
On Mar 17, 2015 8:01 PM, "Jeff Bezanson" <notifications@github.com> wrote:

> In base/deprecated.jl
> <https://github.com/JuliaLang/julia/pull/10543#discussion_r26598614>:
>
> > @@ -496,3 +496,19 @@ function to_index{T<:Real}(A::AbstractArray{T})
> >      depwarn("indexing with non Integer AbstractArrays is deprecated", :to_index)
> >      Int[to_index_nodep(x) for x in A]
> >  end
> > +
> > +function float_isvalid{T<:Union(Float32,Float64)}(s::AbstractString, out::Array{T,1})
> > +    tf = tryparse(T, s)
> > +    isnull(tf) || (out[1] = get(tf))
> > +    !isnull(tf)
> > +end
>
> This is actually just a utility used by actual deprecated functions. There
> never was a float_isvalid before.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JuliaLang/julia/pull/10543/files#r26598614>.
>